packages/react-reconciler/src/ReactFiberWorkLoop.js

```diff
@@ -2453,6 +2454,9 @@ export function warnIfNotCurrentlyActingEffectsInDEV(fiber: Fiber): void {
   if (__DEV__) {
     if (
       warnsIfNotActing === true &&
+      (fiber.mode & StrictMode ||
```

Comment:
should this fire for profiling mode as well? I... guess? will follow up as a separate PR if so.

Reply:
actually I'll get it in here right now, nvm

Reply:
It shouldn't fire if only ProfileMode is set, so this isn't right. It should only fire if either ConcurrentMode or BatchedMode or StrictMode are set. The convention we usually use for bitmask checks is:

```js
if ((mode & (ConcurrentMode | BatchedMode | StrictMode)) !== NoMode) {
  // ...
}

```

ConcurrentMode and BatchedMode are always accompanied by StrictMode is not. So technically it would be sufficient to check only for StrictMode.

```js
if ((mode & StrictMode) !== NoMode) {
  // ...
}
```